app/controllers/homepage_controller.rb

```diff
@@ -21,22 +21,56 @@ def update
 
     flash[:success] = "Popular links draft saved.".html_safe
     redirect_to show_popular_links_path
-  rescue StandardError
+  rescue GdsApi::HTTPErrorResponse
+    flash[:danger] = publishing_api_save_error_message.html_safe
+    render "homepage/popular_links/edit"
+  rescue Mongoid::Errors::Validations
+    render "homepage/popular_links/edit"
+  rescue StandardError => e
+    Rails.logger.error "Error #{e.class} #{e.message}"
+    flash[:danger] = "Due to an application error, the edition couldn't be saved."
     render "homepage/popular_links/edit"
   end
 
   def publish
-    @latest_popular_links.publish
+    @latest_popular_links.publish_latest
+
     flash[:success] = "Popular links successfully published.".html_safe
-    render "homepage/popular_links/show"
+  rescue GdsApi::HTTPErrorResponse => e
+    flash[:danger] = rescue_already_published_error(e)
+  rescue StandardError => e
+    Rails.logger.error "Error #{e.class} #{e.message}"
+    flash[:danger] = "Due to an application error, the edition couldn't be published."
+  ensure
+    redirect_to show_popular_links_path
   end
 
 private
 
+  def rescue_already_published_error(error)
+    already_published_error(JSON.parse(error.http_body)) ? "Popular links publish was unsuccessful, cannot publish an already published content item.".html_safe : publishing_api_publish_error_message
+  end
+
+  def already_published_error(error_body)
+    error_body["error"] && error_body["error"]["message"] && error_body["error"]["message"].include?("already published content item")
+  end
+
   def fetch_latest_popular_link
     @latest_popular_links = PopularLinksEdition.last
   end
 
+  def publishing_api_publish_error_message
+    "Popular links publish was unsuccessful due to a service problem. #{try_again}".html_safe
+  end
+
+  def publishing_api_save_error_message
+    "Popular links save was unsuccessful due to a service problem. #{try_again}".html_safe
+  end
+
+  def try_again
```

Comment:
Maybe we should rename this method (or extract the text to a constant). The current name would suggest to me that it is going to try doing something again, not that it returns a message.